.ibm/pipelines/reporting.sh

```diff
@@ -0,0 +1,133 @@
+  fi
+  echo "${job_complete_url}"
+}
+
+report_ci_slack_alert() {
+  set -x
+  URL_CI_RESULTS=$(get_job_url)
+  local notification_text
+  if [[ $OVERALL_RESULT == 0 ]]; then
+    notification_text=":done-circle-check: \`${JOB_NAME}\`, 📜 <$URL_CI_RESULTS|logs>."
+  else
+    notification_text=':failed: `'"${JOB_NAME}"'`, 📜 <'"$URL_CI_RESULTS"'|logs>, <!subteam^S07BMJ56R8S>.'
+    for ((i = 1; i <= ${#STATUS_DEPLOYMENT_NAMESPACE[@]}; i++)); do
+      URL_ARTIFACTS[i]=$(get_artifacts_url "${STATUS_DEPLOYMENT_NAMESPACE[i]}")
+      URL_PLAYWRIGHT[i]="${URL_ARTIFACTS[i]}/index.html"
+      if [[ "${STATUS_FAILED_TO_DEPLOY[i]}" == "true" ]]; then
+        notification_text="${notification_text}\n• \`${STATUS_DEPLOYMENT_NAMESPACE[i]}\` :circleci-fail: failed to deploy, "
+      else
+        notification_text="${notification_text}\n• \`${STATUS_DEPLOYMENT_NAMESPACE[i]}\` :deployments: deployed, "
+        if [[ "${STATUS_TEST_FAILED[i]}" == "true" ]]; then
+          notification_text="${notification_text}:circleci-fail: test failed, "
+        else
+          notification_text="${notification_text}:circleci-pass: test passed, "
+        fi
+        notification_text="${notification_text}:playwright: <${URL_PLAYWRIGHT[i]}|Playwright>, "
+        if [[ "${URL_REPORTPORTAL[i]}" != "" ]]; then
+          notification_text="${notification_text}:reportportal: <${URL_REPORTPORTAL[i]}|ReportPortal>, "
+        fi
+      fi
+      notification_text="${notification_text}📦 <${URL_ARTIFACTS[i]}|artifacts>."
+    done
+  fi
+  set +x
+
+  echo "${notification_text}"
+  printf "%s" "${notification_text}"
+  echo "Saving the notification text to a file, which is then picked up by a separate CI step."
+  # It is important to save the file in the shared directory with the exact name.
+  echo "${notification_text}" > "${SHARED_DIR}/ci-slack-alert.txt"
+
+  curl -X POST -H 'Content-type: application/json' \
+    --data "{\"text\":\"${notification_text}\"}" \
+    "$SLACK_NIGHTLY_WEBHOOK_URL"
+}
+
+reportportal_slack_alert() {
+  local release_name=$1
+  local reportportal_launch_url=$2
+
+  if [[ "$release_name" == *rbac* ]]; then
+    RUN_TYPE="rbac-nightly"
+  else
+    RUN_TYPE="nightly"
+  fi
+  if [[ ${RESULT} -eq 0 ]]; then
+    RUN_STATUS_EMOJI=":done-circle-check:"
+    RUN_STATUS="passed"
+  else
+    RUN_STATUS_EMOJI=":failed:"
+    RUN_STATUS="failed"
+  fi
+  jq -n \
+    --arg run_status "$RUN_STATUS" \
+    --arg run_type "$RUN_TYPE" \
+    --arg reportportal_launch_url "$reportportal_launch_url" \
+    --arg job_name "$JOB_NAME" \
+    --arg run_status_emoji "$RUN_STATUS_EMOJI" \
+    '{
+      "RUN_STATUS": $run_status,
+      "RUN_TYPE": $run_type,
+      "REPORTPORTAL_LAUNCH_URL": $reportportal_launch_url,
+      "JOB_NAME": $job_name,
+      "RUN_STATUS_EMOJI": $run_status_emoji
+    }' > /tmp/data_router_slack_message.json
+  curl -X POST -H 'Content-type: application/json' --data @/tmp/data_router_slack_message.json  $SLACK_DATA_ROUTER_WEBHOOK_URL
```

Comment:
```suggestion
 if ! curl -X POST -H 'Content-type: application/json' --data @/tmp/data_router_slack_message.json $SLACK_DATA_ROUTER_WEBHOOK_URL; then
    echo "Failed to send Slack notification"
fi
```
Do the same for the other `curl`